presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -107,6 +161,44 @@ public static double ceiling(@SqlType(StandardTypes.DOUBLE) double num)
         return Math.ceil(num);
     }
 
+    private static SqlScalarFunction decimalCeilingFunction(String name)
+    {
+        Signature signature = Signature.builder()
+                .kind(SCALAR)
+                .name(name)
+                .literalParameters("num_precision", "num_scale")
+                .argumentTypes("decimal(num_precision, num_scale)")
+                .returnType("decimal(num_precision, num_scale)")
+                .build();
+        return SqlScalarFunction.builder(MathFunctions.class)
+                .signature(signature)
+                .methods("ceilingShortDecimal", "ceilingLongDecimal")
+                .extraParameters(MathFunctions::decimalScaleExtraParameters)
+                .build();
+    }
+
+    private static List<Object> decimalScaleExtraParameters(SpecializeContext context)
+    {
+        return ImmutableList.of(context.getLiteral("num_scale"));
+    }
+
+    public static long ceilingShortDecimal(long num, long scale)
+    {
+        BigDecimal bigDecimal = BigDecimal.valueOf(num, (int) scale);
```

Comment:
do not create BigDecimal here.
Possibly you also want to pass some derivate of scale as parameter here - to avoid repeated computations for each row.